src/act-compat.js

```diff
@@ -28,19 +28,21 @@ function asyncAct(cb) {
         console.error = function error(...args) {
           /* if console.error fired *with that specific message* */
           /* istanbul ignore next */
-          if (
-            args[0].indexOf(
-              'Warning: Do not await the result of calling ReactTestUtils.act',
-            ) === 0
-          ) {
-            // v16.8.6
-            isAsyncActSupported = false
-          } else if (
-            args[0].indexOf(
-              'Warning: The callback passed to ReactTestUtils.act(...) function must not return anything',
-            ) === 0
-          ) {
-            // no-op
+          if (typeof args[0] === 'string') {
+            if (
+              args[0].indexOf(
+                'Warning: Do not await the result of calling ReactTestUtils.act',
+              ) === 0
+            ) {
+              // v16.8.6
+              isAsyncActSupported = false
+            } else if (
+              args[0].indexOf(
+                'Warning: The callback passed to ReactTestUtils.act(...) function must not return anything',
+              ) === 0
+            ) {
+              // no-op
+            }
```

Comment:
```suggestion
            } else {
              originalConsoleError.apply(console, args)
            }
```

I think that should do it.